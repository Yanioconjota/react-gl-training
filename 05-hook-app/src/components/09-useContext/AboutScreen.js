import React, { useContext } from 'react';
import { UserContext } from "./UserContext";

export const AboutScreen = () => {
  const { user, setUser } = useContext(UserContext);

  const handleClick = () => setUser({});
  
  return (
    <>
      <h1>About screen</h1>
      <hr />
      <pre className="Code mb-3">{JSON.stringify(user, null, 3)}</pre>
      <button
        className="btn btn-secondary"
        onClick={handleClick}>Logout</button>
    </>
  );
};
