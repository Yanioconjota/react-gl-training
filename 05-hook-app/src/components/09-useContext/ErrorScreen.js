import React from 'react';
import kenobi from '../../assets/img/kenobi.gif';

export const ErrorScreen = () => {
  return (
    <div className="d-flex flex-column align-items-center">
      <img className="mt-5" src={kenobi} alt="" />
      <p className="text-yellow">This is not the screen you're looking for</p> 
    </div>
  );
};
