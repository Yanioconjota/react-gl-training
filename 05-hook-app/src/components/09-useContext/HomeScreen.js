import React, { useContext } from "react";
import { UserContext } from './UserContext';

export const HomeScreen = () => {

  const { user } = useContext(UserContext);
  console.log({ user });

  return (
    <>
      <h1>Home screen</h1>
      <hr />
      <pre className="Code">{JSON.stringify(user, null, 3)}</pre>
    </>
  );
};
