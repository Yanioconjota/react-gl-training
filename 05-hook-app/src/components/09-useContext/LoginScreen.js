import React, { useContext } from "react";
import { UserContext } from "./UserContext";

export const LoginScreen = () => {
  const { setUser } = useContext(UserContext);

  const loggedUser = {
    id: 666,
    name: 'Homero',
    lastname: 'Thompson',
    email: 'homerothompson@email.com'
  };

  return (
    <>
      <h1>Login screen</h1>
      <hr />
      <button
        className="btn btn-primary"
        onClick={()=> setUser(loggedUser)}>Login</button>
    </>
  );
};
