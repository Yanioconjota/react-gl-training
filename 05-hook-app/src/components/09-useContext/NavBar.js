import React from 'react';
import { Link, NavLink } from "react-router-dom";

const routes = [
  { path: "/", name: "Home" },
  { path: "/login", name: "Login" },
  { path: "/about", name: "About" },
];

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-sm">
      <span className="navbar-brand">
        <Link className="nav-link" to="/">
          Use Context App
        </Link>
      </span>
      <div className="collapse navbar-collapse" id="navbarNav">
        <ul className="navbar-nav">
          {routes.map((route) => (
            <li className="nav-item" key={route.name}>
              <NavLink
                activeclassname="active"
                className="nav-link" to={route.path}>
                {route.name}
              </NavLink>
            </li>
          ))}
        </ul>
      </div>
    </nav>
  );
};