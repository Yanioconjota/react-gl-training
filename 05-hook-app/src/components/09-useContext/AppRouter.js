import React from 'react';
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { ErrorScreen } from './ErrorScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

//la version 6 cambi√≥ el Switch por el Routes, y component por element.
/*
<Switch>
  <Route path="/about" component={AboutScreen}/>
</Switch>

<Routes>
  <Route path="/about" element={<AboutScreen/>} />
</Routes>
*/

export const AppRouter = () => {
  return (
    <Router>
      <div>
        <NavBar />
        <Routes>
          <Route exact path="/" element={<HomeScreen />} />
          <Route exact path="/about" element={<AboutScreen />} />
          <Route exact path="/login" element={<LoginScreen />} />
          <Route path="*" element={<ErrorScreen />} />
        </Routes>
      </div>
    </Router>
  );
};
