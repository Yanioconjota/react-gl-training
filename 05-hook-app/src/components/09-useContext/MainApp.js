import React, { useState } from 'react';
import { AppRouter } from './AppRouter';
import '../../assets/scss/UseContextApp.scss';
import { UserContext } from './UserContext';

const MainApp = () => {

  // const user = {
  //   id: 666,
  //   name: 'Homero',
  //   email: 'homerothompson@correo.com'
  // };

  const [user, setUser] = useState({});

  return (
    <UserContext.Provider value={{
      user,
      setUser,
      temporal:666
    }}>
      <AppRouter />
    </UserContext.Provider>
  );
};

export default MainApp;
