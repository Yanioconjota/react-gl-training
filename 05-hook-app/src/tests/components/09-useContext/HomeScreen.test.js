import "@testing-library/jest-dom";
import { mount, shallow } from "enzyme";
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from "../../../components/09-useContext/UserContext";

describe('pruebas con HomeScreen', () => {

  const user = {
    id: 666,
    name: 'Homero',
    lastname: 'Thompson',
    email: 'homerothompson@email.com'
  }

  const wrapper = mount(
    <UserContext.Provider value={{
      user
    }}>
      <HomeScreen/>
    </UserContext.Provider>
  );
  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
});
