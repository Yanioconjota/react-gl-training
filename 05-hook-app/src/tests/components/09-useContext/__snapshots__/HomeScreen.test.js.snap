// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas con HomeScreen debe mostrar el componente correctamente 1`] = `
Array [
  <h1>
    Home screen
  </h1>,
  <hr />,
  <pre
    className="Code"
  >
    {
   "id": 666,
   "name": "Homero",
   "lastname": "Thompson",
   "email": "homerothompson@email.com"
}
  </pre>,
]
`;
